static/src/javascripts/projects/common/modules/ui/sticky-social.js

```diff
@@ -40,48 +37,80 @@ define([
     }
 
     function determineStickiness() {
-        if (failed) {
+        if (inited) {
+            setStickiness();
+
+        } else if (!topEl() || !bottomEl()) {
             return;
 
-        } else if (!bottomEl) {
+        } else {
             fastdom.read(function () {
-                bottomEl = $(selectorBottomEl);
-
-                if (bottomEl) {
-                    fastdom.write(function () {
-                        bottomEl.addClass(stickyClassName);
-                        setTimeout(makeRevealable);
-                    });
-                } else {
-                    failed = true;
-                }
+                fastdom.write(function () {
+                    $(bottomEl()).addClass(stickyClassName);
+                    setTimeout(makeRevealable);
+                    inited = true;
+                });
             });
-
-        } else {
-            setStickiness();
         }
     }
 
     function makeRevealable() {
-        fastdom.write(function () { bottomEl.addClass(stickyRevealableClassName); });
+        fastdom.write(function () { $(bottomEl()).addClass(stickyRevealableClassName); });
     }
 
     function reveal() {
         if (!revealed) {
             revealed = true;
-            fastdom.write(function () { bottomEl.addClass(stickyRevealClassName); });
+            fastdom.write(function () { $(bottomEl()).addClass(stickyRevealClassName); });
         }
     }
 
     function unreveal() {
         if (revealed) {
             revealed = false;
-            fastdom.write(function () { bottomEl.removeClass(stickyRevealClassName); });
+            fastdom.write(function () { $(bottomEl()).removeClass(stickyRevealClassName); });
         }
     }
 
+    function moveToFirstPosition($el) {
+        $el.parent().prepend($el.detach());
+    }
+
     function init() {
-        mediator.on('window:scroll', _.throttle(determineStickiness, 10));
+        var testVariant = ab.getTestVariant('ShareButtons'),
+            referrer,
+            socialReferrer;
+
+        if (testVariant.indexOf('referrer') > -1) {
+            referrer = ((window.location.hash + '').match(/referrer=([^&]+)/) || [])[1] || document.referrer,
+
+            socialReferrer = referrer ? [
+                'facebook',
+                'twitter'
+            ].filter(function (social) {
+                return referrer.indexOf(social) > -1;
+            })[0] : null;
+
+            if (socialReferrer) {
+                fastdom.read(function () {
```

Comment:
There is, because the `topEl()` in the block is a fn that potentially reads the dom (if not yet memoized).